tests/generic_config_updater/test_portchannel_interface.py

```diff
@@ -0,0 +1,348 @@
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        for portchannel_name, ips in T0_PORTCHANNEL_TABLE.items():
+            check_show_ip_intf(duthost, portchannel_name, [], [ips['ip']], is_ipv4=True)
+            check_show_ip_intf(duthost, portchannel_name, [], [ips['ipv6']], is_ipv4=False)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def verify_attr_change(duthost, name, attr, value):
+    """
+    attr:
+        mtu: check if "mtu 3324" exists
+            admin@vlab-01:~$ show interfaces status | grep -w ^PortChannel0001
+            PortChannel0001              N/A      40G   3324    N/A             N/A           routed      up       up     N/A         N/A
+        min_links:
+            TODO: further check
+        admin_status: check if 3rd column start with "down"
+            admin@vlab-01:~/lag$ show ip interfaces
+            Interface        Master    IPv4 address/mask    Admin/Oper    BGP Neighbor    Neighbor IP
+            ---------------  --------  -------------------  ------------  --------------  -------------
+            ...
+            PortChannel0001            10.0.0.56/31         up/up         ARISTA01T1      10.0.0.57
+            ...
+    """
+    if attr == "mtu":
+        output = duthost.shell("show interfaces status | grep -w '^{}' | awk '{{print $4}}'".format(name))
+
+        pytest_assert(output['stdout'] == value,
+            "{} attribute {} failed to change to {}".format(name, attr, value)
+        )
+    elif attr == "min_links":
+        pass
+    elif attr == "admin_status":
+        output = duthost.shell("show ip interfaces | grep -w '{}' | awk '{{print $3}}'".format(name))
+
+        pytest_assert(output['stdout'].startswith(value),
+            "{} {} change failed".format(name, attr)
+        )
+
+@pytest.mark.parametrize("op, name, attr, value", [
+    ("replace", "PortChannel0001", "mtu", "3324"),
+    ("replace", "PortChannel0001", "min_links", "2"),
+    ("replace", "PortChannel0001", "admin_status", "down")
+])
+def test_portchannel_interface_tc5_modify_attribute(duthost, op, name, attr, value):
+    """Test PortChannelXXXX attribute change
+
+    ("replace", "PortChannel0001", "mtu", "3324"), mtu change
+    ("replace", "PortChannel0001", "min_links", "2"), min_link change
+    ("replace", "PortChannel0001", "admin_status", "down"), admin_status change
+    """
+    json_patch = [
+        {
+            "op": "{}".format(op),
+            "path": "/PORTCHANNEL/{}/{}".format(name, attr),
+            "value": "{}".format(value)
+        }
+    ]
+
+    logger.info("json patch {}".format(json_patch))
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        verify_attr_change(duthost, name, attr, value)
```

Comment:
>verify_attr_change [](http://example.com/codeflow?start=8&length=18)

For end-to-end test, we could also verify that teamd processes (per port channel) are running without crash.

Reply:
How about checking portchannel pid?
docker exec -it teamd test -f /var/run/teamd/PortChannel0001.pid

Reply:
Checking process such as command `ps` is better. There is risk that the process crashed but the pid file is still there.

Reply:
Changed to `state dump` verification